src/network.cc

```diff
@@ -29,6 +33,17 @@ QTcpSocket *tcpSocket =
 
   tcpSocket->connectToHost(ip, port);
 
+  QTime timer;
+  timer.start();
+  while (!tcpSocket->waitForConnected()) {
+    tcpSocket->connectToHost(ip, port);
+    // Check elapsed time
+    const int time_waiting_ms = 1000;
+    if (timer.elapsed() > time_waiting_ms){
+      fprintf(stderr, "%s\n%s\n", "Error: Connection not established", strerror(errno));
```

Comment:
I think error is going to be stored in `QTCPSocket::errorString` instead of `errno`.